tests/docker_test.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# This script logs into the Datalab service, runs a specified SQL query asynchronously,
+# waits for the query to complete, and then checks if the resulting CSV file contains
+# the expected number of lines (excluding the header).
+
+# Arguments:
+#   $1 (USER_NAME): Username for Datalab login. Mandatory.
+#   $2 (PASSWD): Password for Datalab login. If not provided, the script will asks you
+#                for one.
+
+# Usage:
+#   ./script_name.sh <USER_NAME> [<PASSWD>]
+
+# The script performs the following operations:
+#   1. Logs into Datalab with the provided username and optional password.
+#   2. Executes a predefined SQL query to fetch data from the GAIA DR3 source,
+#      limiting the results to a specified number of lines.
+#   3. Waits for the query to complete by periodically checking the job status.
+#   4. Saves the query results to a CSV file named after the job ID.
+#   5. Checks if the CSV file contains the expected number of lines (excluding the header)
+#      and prints a message indicating whether the check was successful.
+
+set -e
+
+USER_NAME=$1
+PASSWD=$2
+
+if [ -z "${PASSWD}" ]
+then
+  datalab login --user="${USER_NAME}"
+else
+  datalab login --user="${USER_NAME}" --password="${PASSWD}"
+fi
+
+# The expected number of lines
+expected_lines=100000
+export SQL="SELECT ra, dec, parallax, pmra, pmdec, radial_velocity
+FROM gaia_dr3.gaia_source
+WHERE 't' = Q3C_RADIAL_QUERY(ra, dec, 16.95, -4.1, 21.48)
+LIMIT ${expected_lines}
+"
+
+JOBID=$(datalab query --sql="$SQL" --async=True)
+echo "Job ID for submitted async SQL: $JOBID"
+
+STATUS=$(datalab qstatus --jobId="$JOBID")
+while [[ "$STATUS" != "COMPLETED" ]]
+do
+  sleep 5 # Avoids querying the status too frequently
+  STATUS=$(datalab qstatus --jobId="$JOBID")
+  echo "Current status of async query: $STATUS"
+done
+
+
+csv_file=${JOBID}.csv
+echo "saving results to filename:${csv_file}"
+datalab qresults --jobId="$JOBID" --fname=${csv_file}
+
+
+# Count the number of lines excluding the header
+line_count=$(tail -n +2 "$csv_file" | wc -l)
```

Comment:
This looks like it could be slower than just ``line_count=$((`time wc -l < $csv_file`-1))``
(some local testing seems to suggest that)

Reply:
but it is more beautiful! and if beauty costs 0.0000000001% extra of CPU, so be it! LoL!

Reply:
Please test yourself, but I get a factor ~3 in speed difference on an ~800 MB CSV file...